src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -78,7 +79,18 @@ trait BackendCall {
    * expression `read_lines(my_file_var)` would have to call lookupFunction()("my_file_var")
    * during expression evaluation
    */
-  def lookupFunction: String => WdlValue
+  def lookupFunction: String => WdlValue = {
+    def resolveParameter(name: String) = Try(locallyQualifiedInputs.getOrElse(name, throw new Throwable(s"variable $name not found")))
+    def resolveDeclaration(lookup: ScopedLookupFunction)(name: String) = key.scope.task.declarations.find(_.name == name) match {
+      case Some(d) => d.expression.map(_.evaluate(lookup, engineFunctions)).getOrElse(Failure(new WdlExpressionException(s"Could not evaluate declaration $d")))
+      case None => Failure(new Throwable(s"variable $name not found"))
+    }
+    def lookup(key: String): WdlValue = {
+      val attemptedResolutions = Stream(resolveParameter _, resolveDeclaration(lookupFunction) _) map {_(key)} find {_.isSuccess}
+      attemptedResolutions.getOrElse(throw new Throwable(s"variable $key not found")).get
```

Comment:
What miguel said. Also I need a lot of convincing that it's impossible to combine these somehow.